datafusion/physical-expr/src/string_expressions.rs

```diff
@@ -303,15 +303,29 @@ pub fn instr<T: OffsetSizeTrait>(args: &[ArrayRef]) -> Result<ArrayRef> {
     let string_array = as_generic_string_array::<T>(&args[0])?;
     let substr_array = as_generic_string_array::<T>(&args[1])?;
 
+    fn process_instr(string: &str, substr: &str) -> Option<i64> {
```

Comment:
this is likely substantially slower than the previous implementation as it copies the strings into Vecs on each row